dialog/file.go

```diff
@@ -181,10 +184,36 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
 
 	favorites := f.loadFavorites()
 
-	favoritesGroup := widget.NewGroup("Favorites", favorites...)
-	return fyne.NewContainerWithLayout(layout.NewBorderLayout(header, footer, favoritesGroup, nil),
-		favoritesGroup, header, footer, body)
+	favoritesGroup := widget.NewVScrollContainer(widget.NewGroup("Favorites", favorites...))
+	var optionsButton *widget.Button
+	optionsButton = widget.NewButton("Options", func() {
```

Comment:
```suggestion
	optionsButton := widget.NewButton("Options", func() {
```
You should just be able to create the button in one step without declaring the var first.

Reply:
Nope, the var is needed, as the func needs to reference the options button.

Reply:
Oh, right. Sorry. Don’t know how I could miss that.

Reply:
Erm, the usage of “:=“ will result in the same variable being available for reference...

Reply:
Yes the use of ":=" creates teh same variable as the var line, but it is created after the inline anonymous function is evaluated which results in the compiler error "undefined: optionsButton"